servers/republik/script/prolong/importCancellations.js

```diff
@@ -0,0 +1,97 @@
+/**
+ * This script imports cancellations from a csv
+ * columns: date, adminUrl, reason, category
+ *
+ * Usage:
+ * cat local/cancellations.csv | node script/prolong.js
+ */
+require('@orbiting/backend-modules-env').config()
+const PgDb = require('@orbiting/backend-modules-base/lib/pgdb')
+const rw = require('rw')
+const {dsvFormat} = require('d3-dsv')
+const csvParse = dsvFormat(',').parse
+const moment = require('moment')
+
+console.log('running importCancellations.js...')
+
+const getNormalizedCategory = (category) => {
+  switch (category) {
+    case 'Inhaltliche Ausrichtung':
+      return 'EDITORIAL'
+    case 'Keine Zeit':
+      return 'NO_TIME'
+    case 'Zu teuer':
+      return 'TOO_EXPENSIVE'
+    case 'keine':
+      return 'VOID'
+    default:
+      return 'OTHER'
+  }
+}
+
+PgDb.connect().then(async pgdb => {
+  const dry = process.argv[2] === '--dry'
+  if (dry) {
+    console.log("dry run: this won't change anything")
+  }
+
+  const input = rw.readFileSync('/dev/stdin', 'utf8')
+  if (!input || input.length < 4) {
+    throw new Error('You need to provide input on stdin')
+  }
+
+  let stats = {
+    numMemberships: 0
+  }
+  const numCancellationsBefore = await pgdb.public.membershipCancellations.count()
+
+  const transaction = await pgdb.transactionBegin()
+  await Promise.all(
```

Comment:
(Maybe use bluebird's `Promise.map`, on which you can steer concurrency. `Promise.all` will execute in parallel.)